kedro/io/data_catalog.py

```diff
@@ -436,16 +439,21 @@ def __contains__(self, data_set_name):
             return True
         return False
 
+    @classmethod
     def _resolve_config(
-        self,
+        cls,
         data_set_name: str,
         matched_pattern: str,
+        config_copy: dict,
     ) -> dict[str, Any]:
         """Get resolved AbstractDataset from a factory config"""
         result = parse(matched_pattern, data_set_name)
-        config_copy = copy.deepcopy(self._dataset_patterns[matched_pattern])
         # Resolve the factory config for the dataset
         for key, value in config_copy.items():
+            if isinstance(value, Dict):
```

Comment:
Can we keep `dict` here? Unsure why we need typing here.

Reply:
fixed!